### ISSUE TITLE
feature request/legacy ghc support on future OS X: expose appending to ghc override options in ~/.cabal/config

### ISSUE THREAD

**Issue by cartazio** [haskell/cabal#1383]
to repeat the title:
feature request/legacy ghc support on future OS X: expose appending to ghc override options in ~/.cabal/config as another field (to allow the cabal user to globally specify things like the values for -pgma -pgmc and -pgmP flags, among others)

Come OS X 10.9 and the associated dev tools, there will be no apple provided GCC, just something that is "named" gcc, but actually is clang. 

GHC HEAD (thus GHC 7.8 onwards) have been patched so that a suitably recent Clang will work as a full GCC substitute. 

However, ghc <= 7.6.3 needs GCC for the CPP pass, and can't use Clang for that role. Naively, one might think that building GCC on such a mac, and adjusting the ghc  SETTINGS file to point to custom built GCC as its C compiler would suffice. It doesn't and can result in gnarly breakage and weird linking problems on a system who's default C/C++ compiler is clang, while GHC is off using gcc/g++.

However, consider the scenario of linking to a library written using C++, which was probably built using Clang (being the default C/C++ compiler), with some additional C++ as part of a haskell library being used to provide a wrapper with an extern "C" interface that the haskell code itself links to. Using the naive solution of pointing GHC to using GCC (not the apple provided one which will be no more) for all of its passing, assembling, compiling, linking will yield odd linking errors, see   this closed ticket on llvm-general for an example https://github.com/bscarlet/llvm-general/issues/24

basically: one work around for the end user on 10.9 that is simplest, for enabling effective usage of ghc < 7.7 on pending OS X 10.9, would be allowing the user in their ~/.cabal/config file to have some sort of `ghc-options-override:` flag exposed.

in which case, the way I would hypothetically use it would be 
`ghc-options-override: -pgmc clang -pgma clang -pgml clang -pgmP gcc` to have working builds.

the specifics of the cabal configuration information i'm referring to is the same as that which i'm manipulating in this linked setup.hs example file https://gist.github.com/cartazio/5842786#file-setup-hs-L39

thanks
